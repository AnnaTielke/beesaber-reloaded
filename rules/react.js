module.exports = {
  plugins: ['react'],

  extends: ['plugin:react/recommended'],

  settings: {
    react: {
      version: 'detect',
    },
  },

  rules: {
    'react/button-has-type': 'error',
    'react/jsx-boolean-value': 'error',
    'react/jsx-fragments': 'error',
    'react/jsx-handler-names': 'error',
    'react/jsx-no-bind': ['error', { allowArrowFunctions: true }],
    'react/jsx-pascal-case': 'error',
    'react/no-access-state-in-setstate': 'error',
    'react/no-array-index-key': 'error',
    'react/no-danger': 'error',
    'react/no-did-mount-set-state': 'error',
    'react/no-did-update-set-state': 'error',
    'react/no-multi-comp': 'error',
    'react/no-this-in-sfc': 'error',
    'react/no-typos': 'error',
    'react/no-unsafe': 'error',
    'react/no-unused-prop-types': 'error',
    'react/no-unused-state': 'error',
    'react/no-will-update-set-state': 'error',
    'react/prefer-es6-class': 'error',
    'react/prefer-stateless-function': 'error',
    'react/style-prop-object': 'error',
    'react/void-dom-elements-no-children': 'error',
  },
};
