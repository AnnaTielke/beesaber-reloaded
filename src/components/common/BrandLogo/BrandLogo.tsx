import React from 'react';
import styled from 'styled-components';

import tokens from '@/tokens/index';

import BrandLogoI from './BrandLogoI/BrandLogoI';

const Container = styled.div`
  position: relative;
  height: 150px;
  width: 150px;
  font-size: 80px;
  margin-bottom: 1rem;

  div {
    position: absolute;
    transform-origin: center;
  }
  div:first-child {
    transform: rotate(90deg) translate(-36px, -68px);
    z-index: 5;
  }
  div:nth-child(2) {
    transform: rotate(150deg) translate(-103px, -49px);
    z-index: 4;
  }
  div:nth-child(3) {
    transform: rotate(30deg) translate(120px, -19px);
  }
  div:nth-child(4) {
    transform: rotate(90deg) translate(70px, -68px);
  }
  div:nth-child(5) {
    transform: rotate(150deg) translate(2px, -49px);
  }
  div:nth-child(6) {
    transform: rotate(30deg) translate(15px, -20px);
    z-index: 6;
  }

  .saber {
    position: absolute;
    height: 100%;
    width: 3px;
    top: 0%;
    border-radius: 3px;
  }

  .saber.neon-red {
    left: 49%;
    transform: rotate(30deg);
    background-color: ${tokens.color.red.light};
    box-shadow: 0px 0px 8px 2px ${tokens.color.red.shadow},
      0px 0px 12px 6px ${tokens.color.red.main};
  }

  .saber.neon-blue {
    right: 49%;
    transform: rotate(-30deg);
    background-color: ${tokens.color.blue.light};
    box-shadow: 0px 0px 8px 2px ${tokens.color.red.shadow},
      0px 0px 12px 6px ${tokens.color.red.main};
  }
`;

const BrandLogo = (): JSX.Element => (
  <Container>
    {/* neon honeycomb logo: */}
    <BrandLogoI />
    <BrandLogoI />
    <BrandLogoI />
    <BrandLogoI />
    <BrandLogoI />
    <BrandLogoI />
    {/* alternative logo made with svg: */}
    {/* <svg className="logo"><use xlinkHref="#logo-honeycomb" /></svg> */}
  </Container>
);

export default BrandLogo;
