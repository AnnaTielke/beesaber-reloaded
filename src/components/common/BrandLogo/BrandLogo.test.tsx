import { render } from '@testing-library/react';
import React from 'react';

import BrandLogo from './BrandLogo';

describe('BrandLogo', () => {
  it('should match the snapshot', () => {
    const { container } = render(<BrandLogo />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
