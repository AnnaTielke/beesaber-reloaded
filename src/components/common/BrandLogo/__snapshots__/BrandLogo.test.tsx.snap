// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandLogo should match the snapshot 1`] = `
<div
  class="sc-dkPtRN letjLb"
>
  <div
    class="sc-bdvvtL etKHLZ"
  >
    <span
      class="sc-gsDKAQ fDJHTz"
    >
      I
    </span>
  </div>
  <div
    class="sc-bdvvtL etKHLZ"
  >
    <span
      class="sc-gsDKAQ fDJHTz"
    >
      I
    </span>
  </div>
  <div
    class="sc-bdvvtL etKHLZ"
  >
    <span
      class="sc-gsDKAQ fDJHTz"
    >
      I
    </span>
  </div>
  <div
    class="sc-bdvvtL etKHLZ"
  >
    <span
      class="sc-gsDKAQ fDJHTz"
    >
      I
    </span>
  </div>
  <div
    class="sc-bdvvtL etKHLZ"
  >
    <span
      class="sc-gsDKAQ fDJHTz"
    >
      I
    </span>
  </div>
  <div
    class="sc-bdvvtL etKHLZ"
  >
    <span
      class="sc-gsDKAQ fDJHTz"
    >
      I
    </span>
  </div>
</div>
`;
